massa-execution-worker/src/speculative_ledger.rs

```diff
@@ -360,6 +361,55 @@ impl SpeculativeLedger {
         Ok(())
     }
 
+    /// Gets a copy of a datastore keys for a given address
+    ///
+    /// # Arguments
+    /// * `addr`: address to query
+    ///
+    /// # Returns
+    /// `Some(Vec<Vec<u8>>)` for found keys, `None` if the address does not exist.
+    pub fn get_keys(&self, addr: &Address) -> Option<BTreeSet<Vec<u8>>> {
+
+        let read_guard = self.final_state.read();
+        let mut keys: Option<BTreeSet<Vec<u8>>> = match read_guard.ledger.entry_exists(addr) {
+            true => read_guard.ledger.get_datastore_keys(addr),
```

Comment:
I think a more logical way would be to have the final state `get_datastore_keys` do the entry_exists check and return None if the entry doesn't exist, instead of always returning Some

Reply:
@sydhds is related to this https://github.com/massalabs/massa/issues/3275 ?

Reply:
https://github.com/massalabs/massa/issues/3275 is a followup to improve the performance of get_datastore_keys, here it is about the function actually doing what it is supposed to